web/src/lib/utils/navigation.ts

```diff
@@ -1,3 +1,82 @@
+import { goto } from '$app/navigation';
+import { page } from '$app/stores';
+import { AppRoute } from '$lib/constants';
+import { getAssetInfo } from '@immich/sdk';
+import type { NavigationTarget } from '@sveltejs/kit';
+import { get } from 'svelte/store';
+
 export const isExternalUrl = (url: string): boolean => {
   return new URL(url, window.location.href).origin !== window.location.origin;
 };
+
+export const isPhotosRoute = (route?: string | null) => route?.startsWith('/(user)/photos/[[assetId=id]]') || false;
+export const isSharedLinkRoute = (route?: string | null) => route?.startsWith('/(user)/share/[key]') || false;
+export const isSearchRoute = (route?: string | null) => route?.startsWith('/(user)/search') || false;
+export const isAlbumsRoute = (route?: string | null) => route?.startsWith('/(user)/albums/[albumId=id]') || false;
+export const isPeopleRoute = (route?: string | null) => route?.startsWith('/(user)/people/[personId]') || false;
```

Comment:
You could remove all those `|| false` and instead write `!!route?.startsWith(...)`. Probably up to personal preference though